Architecture.md

```diff
@@ -0,0 +1,100 @@
+# Architecture
+
+## High Level
+
+| Entry Point | Game Loop | Outputs        |
+|-------------|-----------|--------------: |
+| main.js     | game.js   | SIG_TERM, Save |
+
+### `main.js`
+
+Serves as a boostrapping file for the environment the game will run in.
+
+Provides to `game.js`:
+
+- EventBus -- backbone of the physical game loop. Must be synchronous.
+- GameState -- a clean one for a new game, else a dirty one for a loaded save file.
+- userInterface -- an interface bridging the game to an arbitrary UI to describe it.
```

Comment:
Wondering, for consistency sake, if we should rename that from `userInterface` -> `UserInterface`?